llvm/unittests/Frontend/OpenMPIRBuilderTest.cpp

```diff
@@ -5301,6 +5319,23 @@ TEST_F(OpenMPIRBuilderTest, TargetRegionDevice) {
       Constant::getNullValue(PointerType::get(Ctx, 0)),
       Constant::getNullValue(PointerType::get(Ctx, 0))};
 
+  auto SimpleArgAccessorCB = [&](llvm::Argument &Arg, llvm::Value *Input,
+                                 IRBuilderBase &Builder) {
```

Comment:
Should this callback also take an AllocaIP? Without that, we make it rely on the caller to make sure builder is set at the right IP.

Reply:
It's currently only ever called by createOutlinedFunction, which does pre-set the insertion point (and currently doesn't use the AllocaIP which results in slightly differing IR to Clang currently) and takes a lot of ownership of where that insertion point is at the moment, so this mainly just keeps the status quo that exists right now. 

However, I am more than happy to allow it to take an IP or perhaps two, to make the IR from MLIR vs Clang look a little more identical and this would also allow me to utilise the ModuleTranslation builder as @TIFitis suggests

Reply:
I don't think differing IRs is a problem as long as they are correct. I have not worked much on target side, so I am okay with the current implementation and would also be okay with adding an IP, whatever works best.